pkg/operator/ceph/cr_manager.go

```diff
@@ -20,18 +20,46 @@ import (
 	"context"
 
 	"github.com/pkg/errors"
+	cephv1 "github.com/rook/rook/pkg/apis/ceph.rook.io/v1"
+	"github.com/rook/rook/pkg/apis/rook.io/v1alpha2"
 	"github.com/rook/rook/pkg/operator/ceph/cluster"
 	"github.com/rook/rook/pkg/operator/ceph/disruption/controllerconfig"
+	"k8s.io/apimachinery/pkg/runtime"
 
+	mapiv1 "github.com/openshift/cluster-api/pkg/apis/machine/v1beta1"
+	healthchecking "github.com/openshift/machine-api-operator/pkg/apis/healthchecking/v1alpha1"
+	clientgoscheme "k8s.io/client-go/kubernetes/scheme"
 	"sigs.k8s.io/controller-runtime/pkg/client/config"
 	"sigs.k8s.io/controller-runtime/pkg/manager"
 )
 
+var (
+	resourcesSchemeFuncs = []func(*runtime.Scheme) error{
+		clientgoscheme.AddToScheme,
+		mapiv1.AddToScheme,
+		healthchecking.AddToScheme,
+		cephv1.AddToScheme,
+		v1alpha2.AddToScheme,
+	}
+)
+
 func (o *Operator) startManager(namespaceToWatch string, context context.Context, mgrErrorCh chan error) {
+	logger.Info("setting up schemes")
+	// Setup Scheme for all resources
+	scheme := runtime.NewScheme()
+	for _, f := range resourcesSchemeFuncs {
```

Comment:
What is the benefit of creating the schemes in this central place? It seems better to keep it with each type as it was before. If someone creates a new controller they may not see where to add the scheme here.

Reply:
It was useful with the previous version of this patch as the schemes must be defined when building our own cached resources.
I still believe it's nice to have that centralized. If you create a new controller you still need to add the new CRD to `pkg/apis/ceph.rook.io/v1/register.go`.
At the end, these lines will never be touched again and fewer lines will be added in new controllers.

Thoughts?

Reply:
Ok, the functionality is really the same. It's fine either way, we can go with this.